tools/lerno/README.md

```diff
@@ -0,0 +1,37 @@
+lerno
+=====
+
+A caching build tool for NodeJS monorepos.
+
+## Usage
+
+Run this in the root of the repository to do the build, quicker:
+
+```
+tools/lerno/bin/lerno
+```
+
+## How it works
+
+For every package in the monorepo (currently using `lerna ls` to detect the
+packages), figure out the source files and the dependencies, and then hash those
+to determine if the build is stale or a cached result can be reused.
+
+Source files are determined by skipping `node_modules` outright, and
+using `.gitignore` patterns for the rest.
+
+The command that is run is `yarn build+test`, but can be configured
+by passing `--command`.
+
+The entire contents of the package directory (again, except the `node_modules`
```

Comment:
use the term "artifact" to represent the build result (just more prevalent terminology)